app/controllers/lead_providers/partnership_csv_uploads_controller.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module LeadProviders
+  class PartnershipCsvUploadsController < ::LeadProviders::BaseController
+    def new
+      @partnership_csv_upload = PartnershipCsvUpload.new
+    end
+
+    def create
+      if params[:partnership_csv_upload].blank?
+        @partnership_csv_upload = PartnershipCsvUpload.new
+        @partnership_csv_upload.errors[:base] << "Please select a CSV file to upload."
+        render :new and return
+      end
+
+      @partnership_csv_upload = PartnershipCsvUpload.new(upload_params.merge(lead_provider_id: current_user.id))
```

Comment:
Yes, that's right. but both lead_providers and admins have access to this function ? But the model is belongs _to lead_provider...

Reply:
Will admins have access to this? 

If they do, then the model should belong_to user.

Reply:
This would be something like `current_user.lead_provider_profile.lead_provider.id`. Admins will need a different interface for this, since they would need to choose a lead provider

Reply:
I will check this, thanks

Reply:
If signed in as a lead_provider, `current_user.id` and `current_user.lead_provider_profile.lead_provider.id` are the same? I have updated, but should the page only be accessible to lead_providers, and not admins, or both ?